Freelancer Update

ItsDzordz opened the pull request:
# FreeLancer Update
![image](https://github.com/user-attachments/assets/ef0910e5-4fd3-46d3-962f-e0e75334f183)

# Description
This update aims to make Freelancers into its seprate faction that can be used by admins for events, so the station doesn't immediately attack people due to them usign sindie doors for access etc.

New Content:
- 3 armor sets (seprate body and helmet)
- 6 storage containers
- 1 wall type
- 1 vending machine
- 1 job type (non round start)
- 1 acces type
- 4 door types
(for more info check media)

---

# TODO

Phase 1
- [x] Add Freelancer access type
- [x] Add additinal textures for access accsories
- [x] Add Extra door types with freelancer access
- [x] Add Additional hardsuit types 
- [x] Add additional Crate and locker types
- <s>Add Filled lockers with freelancer gear for easy mapping </s>
- [x] Add Mercenery Vending machine
- [x] Fix the freelancer pda texture not showing
Phase 2
- [x] Add Spawnable freelancer ships 
- [x] Add compatibilty textures for the new hardsuits
Phase 3
- [ ] Feedback and polishing
---

<details><summary><h1>Media</h1></summary>
<p>

Armor values are beeing tweaked.

![image](https://github.com/user-attachments/assets/495453ae-9a61-451e-a63c-1049aa10e287) 
![image](https://github.com/user-attachments/assets/4543889b-e994-466f-aeb0-1db4f91f0558) 
---

![image](https://github.com/user-attachments/assets/8d3581d4-69b8-4b05-9662-d054a80baa30)
![image](https://github.com/user-attachments/assets/6f12ee8f-70aa-4c0d-97ee-9a4277055f7c)

---
![image](https://github.com/user-attachments/assets/d367d2f0-228d-470e-a47d-61315d1f6dd5)
![image](https://github.com/user-attachments/assets/b75661ff-4999-4fbc-9a02-3ee1e14fdcba)
---
![image](https://github.com/user-attachments/assets/ce3c9c02-ade8-4977-909e-5c8068273875)
![image](https://github.com/user-attachments/assets/3a86d135-9785-461a-89f6-a42aabb3e080)
![image](https://github.com/user-attachments/assets/05a16f63-1a43-41ec-bdcc-ed0c159eb7a7)
---
![image](https://github.com/user-attachments/assets/70447602-f840-473a-9b2f-81b47427988c)
![image](https://github.com/user-attachments/assets/eab6fafd-d190-4f00-a446-03755d79b6f7)
![image](https://github.com/user-attachments/assets/3511f85d-b9f2-4be5-8f96-7e6375a67efa)
![image](https://github.com/user-attachments/assets/34717097-209e-4dc9-b461-7daaaace6052)
![image](https://github.com/user-attachments/assets/c1bd8b9e-b638-4893-81f2-2817f7b50adb)




</p>
</details>

---

# Changelog

:cl:
- add: Added Freelancer access type
- add: Added Frelancer job type, allowing for quick set outfit
- add: Added additinal textures for access accsories
- add: Added Extra door types with freelancer access
- add: Added additional crates
- add: Added freelancer secure closet
- add: Added freelancer vend
- add: Added 3 hardsuit sets
- add: Added freelancer access docking ports
- fix: fixed broken pda texture
- fix: fixed missing pixels on armor
- add compatibility sprites for resomi and harpy

ItsDzordz committed:
Freelancer update

Add various freelancer structures and new type of access

ItsDzordz committed:
Additional Structures and bug fixes

Added Lockers
Fixed the merc pda

Eagle0600 reviewed (changes requested):
Newlines at end of file should generally have no whitespace in .yml files. It should be newline->end of file.

Eagle0600 commented:
I really like the idea of this. Executing it as a toolset for admins also lets us actually code it into events later if direction approves (travelling chefs having Freelancer access?)

ItsDzordz committed:
Feature update

+ Freelancer icon
+ Freelancer vend
+ freelancer space suits

ItsDzordz committed:
Vending discription update

ItsDzordz committed:
Armor Update

+ race compatibility sprites
+ armor values tweak
+ polished missing pixels from sprites

ItsDzordz committed:
Merge branch 'Fansana:master' into Protocol-3

ItsDzordz committed:
Ships update

+ Freelancer ships and satelites
+ locked windoors

ItsDzordz commented:
i will need help with resolving the current conflicts and adjusting armor values, sine im going o be honest im a bit overwhelmed with the current changes, and i just want to push this content before it breaks even more due to new updates

ItsDzordz committed:
Added simulacrum

adds a modular arena ship called simulacrum

Memeji commented:
I'm still down for helping. I just need way more time than I have at the moment

VividPups commented:
The easiest way I did it because since the Main folder was moved to _Floof now.  is to copy every file you change and did to your desktop and then make a new branch (so it shows the main file change)  and then replace the files that You edited. there is a more proper way appearnlty but I don't know how to really do it

Commit:
Merge branch 'master' into Protocol-3

ItsDzordz committed:
Directory update

moving files to new updated directories

Commit:
Merge branch 'master' into Protocol-3

Commit:
fix tesst

ItsDzordz committed:
job icon fix

ItsDzordz committed:
Merge branch 'Protocol-3' of https://github.com/ItsDzordz/DzordzStuff into Protocol-3

ItsDzordz committed:
Bug fix

ItsDzordz commented:
ok it should work now, but i cant seem to be able to run it on my dev envoirment, no error no anything, client works server booted but just doesnt do anything when i try to connect, i would liek to request for somone to check if it works for them.

Mnemotechnician commented:
> Newlines at end of file should generally have no whitespace in .yml files. It should be newline->end of file.

Not true by the way. Trailing newlines are preferred in all non-binary files due to the specifics of how git calculates file diffs. Whether or not there's whitespace on it doesn't really matter so long as it's kept consistent in future modifications of the file.

ItsDzordz commented:
ok managed to boot it tested the items and it doesnt seem to post any errors anymore

Mnemotechnician commented on Resources/Prototypes/Entities/Objects/Tools/access_configurator.yml:
```diff
@@ -147,3 +147,4 @@
     - CentralCommand
     - NuclearOperative
     - SyndicateAgent
+    - Freelancer
```
```suggestion
    - Freelancer # Floofstation
```

Mnemotechnician commented on Resources/Prototypes/_Floof/Entities/Clothing/Head/misc.yml:
```diff
@@ -41,3 +41,96 @@
+    - HeadSide
+    - HeadTop
+  - type: IngestionBlocker
+  - type: IdentityBlocker
```
Some of these components are defined in the parent and are not neccessary here.
```suggestion
  - type: Sprite
    sprite: _Floof/Clothing/Head/pilothelmet.rsi
  - type: Clothing
    sprite: _Floof/Clothing/Head/pilothelmet.rsi
  - type: Tag
    tags:
    - HelmetEVA
    - WhitelistChameleon
  - type: Armor
    modifiers:
      coefficients:
        Blunt: 0.9
        Slash: 0.9
        Piercing: 0.9
        Heat: 0.9
```

ItsDzordz replied:
Stremlined as requested in newest update.

Mnemotechnician commented on Resources/Prototypes/_Floof/Entities/Clothing/Head/misc.yml:
```diff
@@ -41,3 +41,96 @@
+    - HeadSide
+    - HeadTop
+  - type: IngestionBlocker
+  - type: IdentityBlocker
```
...same thing as above

ItsDzordz replied:
Stremlined as requested in newest update.

Mnemotechnician commented on Resources/Prototypes/_Floof/Entities/Clothing/Head/misc.yml:
```diff
@@ -41,3 +41,96 @@
+    - HeadSide
+    - HeadTop
+  - type: IngestionBlocker
+  - type: IdentityBlocker
```
Again

ItsDzordz replied:
Stremlined as requested in newest update.

Mnemotechnician commented on Resources/Prototypes/_Floof/Entities/Clothing/OuterClothing/hardsuits.yml:
```diff
@@ -12,3 +12,131 @@
+  - type: UserInterface
+    interfaces:
+      enum.StorageUiKey.Key:
+        type: StorageBoundUserInterface
```
This should be inherited either from base outer clothing with storage (which defines the same 3x2 storage grid and will not require any of those components) or be parented from base storage item (in which case you will need to keep the Storage comp).

To add multiple parents, change the `parent` field to an array: `parent: [FirstParent, SecondParent, ...]`. Note that if two parents define the same component, only the first inclusion of the component will make it to the child.

ItsDzordz replied:
Stremlined as requested in newest update.

Mnemotechnician commented on Resources/Prototypes/_Floof/Entities/Clothing/OuterClothing/hardsuits.yml:
```diff
@@ -12,3 +12,131 @@
+  - type: ClothingSpeedModifier
+    walkModifier: 1.0
+    sprintModifier: 1.0
+  - type: HeldSpeedModifier
```
This should be defied in parent.
```suggestion
```

ItsDzordz replied:
Stremlined as requested in newest update.

Mnemotechnician commented on Resources/Prototypes/_Floof/Entities/Clothing/OuterClothing/hardsuits.yml:
```diff
@@ -12,3 +12,131 @@
+  - type: UserInterface
+    interfaces:
+      enum.StorageUiKey.Key:
+        type: StorageBoundUserInterface
```
Same as the above...

ItsDzordz replied:
Stremlined as requested in newest update.

Mnemotechnician commented on Resources/Prototypes/_Floof/Entities/Clothing/OuterClothing/hardsuits.yml:
```diff
@@ -12,3 +12,131 @@
+  - type: UserInterface
+    interfaces:
+      enum.StorageUiKey.Key:
+        type: StorageBoundUserInterface
```
Yes, again, same as the above.

ItsDzordz replied:
Stremlined as requested in newest update.

Mnemotechnician commented on Resources/Prototypes/_Floof/Entities/Structures/Walls/walls.yml:
```diff
@@ -0,0 +1,58 @@
+  - type: StaticPrice
+    price: 150
+  - type: RadiationBlocker
+    resistance: 8
```
Does this really need *any* of those components at all? They all should already be defined in the reinforced wall prototype. You should inherit them from it instead of copy-pasting.

Mnemotechnician reviewed (changes requested):
Several entity prototypes have a lot of redundant/unsafely defined components (saying unsafely because if wizden or upstream decide to butcher the storage system again, all of your unsafe prototypes will break because they do not inherit from the proper bases)

ItsDzordz committed:
Update Resources/Prototypes/Entities/Objects/Tools/access_configurator.yml

Co-authored-by: Mnemotechnican <69920617+Mnemotechnician@users.noreply.github.com>

ItsDzordz committed:
Optimazation Update

streamlined the code based on feedback

Mnemotechnician reviewed (approved):
I think yaml prototypes look fine now. I'll leave map review to South.

Eagle0600 commented on Resources/Prototypes/Floof/Entities/Objects/Devices/pda.yml:
```diff
@@ -17,3 +17,20 @@
+  - type: HealthAnalyzer
+    scanDelay: 1
+    scanningEndSound:
+      path: "/Audio/Items/Medical/healthscanner.ogg"
```
Replace
```
- type: entity
  parent: BasePDA
  id: FreelancerPDA
  name: Freelancer PDA
  description: Refurbished standard PDA
  components:
  - type: HealthAnalyzer
    scanDelay: 1
    scanningEndSound:
      path: "/Audio/Items/Medical/healthscanner.ogg"
```
with 
```
- type: entity
  parent: BaseMedicalPDA
  id: FreelancerPDA
  name: Freelancer PDA
  description: Refurbished standard PDA
  components:
```
This is to ensure that later changes to the function of medical PDAs are inherited. In fact, such changes have already happened since this PR was written.

ItsDzordz replied:
Applied the changes, for some reason it pops up a error when ya equip the pda now, its non faital and i think its has to do with the icon on the toggle menu

Eagle0600 reviewed (changes requested):
In addition the the PDA-specific changes, all /Floof/ files should be moved to /_Floof/, since that directory was renamed.

ItsDzordz committed:
Update pda.yml

Memeji reviewed (approved):
Tested out the functionality and it works locally.
There is two issues not related to the PR, but should be brought up.
1: Access Configurator is broken, crashes when using it.
2: The suits have weird 39.999999~% stamina damage reduction, which is similar to the Radiation suit's issue.

Should be good to go!

southbridge-fur commented on Resources/Maps/Floof/Freelancer/Freelancer_drop_pod.yml:
![image](https://github.com/user-attachments/assets/957debd5-de60-48cd-8e2f-159df8b04cc4)
Please don't stack a substation and APC. You can put the APC on the right wall in the free spot behind the gyroscope.

ItsDzordz replied:
why cant we stack?

southbridge-fur replied:
looks ugly

southbridge-fur commented on Resources/Maps/Floof/Freelancer/Freelancer_drop_pod.yml:
```diff
@@ -0,0 +1,758 @@
+    - type: MetaData
+      name: Emergency Pod
+    - type: Transform
+      pos: 29.40625,-0.484375
```
Change to 0,0

southbridge-fur commented on Resources/Maps/Floof/Freelancer/Freelancer_miner.yml:
Needs a Gyroscope or else you can't turn.

Please don't stack the wall mounted substation on top of the grav generator
![image](https://github.com/user-attachments/assets/de985aff-69ec-4ff0-b020-9cc3daf2bfa7)

---

The left and right lights aren't close enough to the LV wire to be powered, you can extend the LV wire out and up one tile for it to reach both of them

*Unedited*
![image](https://github.com/user-attachments/assets/81d71b8c-a585-4a49-89b4-6b8f84410a1b)
*Edited*
![image](https://github.com/user-attachments/assets/4608b17d-17ea-4c96-8174-c4017437b480)

---

One RTG is not enough to power all this 
![image](https://github.com/user-attachments/assets/7eb0ea6e-1213-480e-a232-572555b7d35e)

---

Monitor frame but no real use for it? 
![image](https://github.com/user-attachments/assets/c6aefbba-d97b-491d-a15d-ca55be66bc91)

---

Also I'd recommend changing the external lights to either Sodium or External ones, since if you're going to do real mining on this you need to light up the rock really well. You could try using some post lights as well to help extend their reach really far out. or just replace the regular external lights with the ones specific for external use. Up to you.

southbridge-fur commented on Resources/Maps/Floof/Freelancer/Freelancer_satelite:
Color all the pipes on a larger area like this. You can easily do it with `colornetwork <pipe entityid> Pipe <color>` the common colorcodes can be found in the [Wizden mapping guide](https://docs.spacestation14.com/en/space-station-14/mapping/guides/general-guide.html#standard-pipe-colors) and here [Resources/Prototypes/Entities/Objects/Tools/spray_painter.yml](https://github.com/Fansana/floofstation1/blob/5bdd15495b2e7fc7a8d94d1ad209ef03fa654e00/Resources/Prototypes/Entities/Objects/Tools/spray_painter.yml)

---

Just so you know, you don't have to put these vacuum markers over lattice, just plating and regular flooring - anything that can hold an atmosphere. 
![image](https://github.com/user-attachments/assets/04e75e60-a289-44a2-be8d-32cb794c2080)

---

If this is intended to move, I recommend not including a station anchor at all.
![image](https://github.com/user-attachments/assets/c94fec0c-df3d-46c4-bca3-9ad43aa429e1)

---
Fax machine is unnamed 
![image](https://github.com/user-attachments/assets/55eaebae-34db-4bd9-a6cd-6a48f3bf4830)

---
Idk how I feel about insuls being easy to obtain for freelancers. If they're supposed to be salvagers I guess it doesn't matter since they're easy to find. That being said I'd probably remove the pair here since the hydro equipment crate comes with gloves.
![image](https://github.com/user-attachments/assets/01aa99b9-14f4-4af7-95f9-c6beeea90410)

---
Not sure how I feel about these many lights here, it's pretty bright
![image](https://github.com/user-attachments/assets/97d82579-e566-482b-b743-80266c5cf5ca)


---
Honestly just give them a filled dispenser 
![image](https://github.com/user-attachments/assets/5a289994-88e3-4c85-a986-7b45987e805b)

---
Why put the lockers in the torture room?
![image](https://github.com/user-attachments/assets/67652fb6-75a3-4083-94a3-3f6d565cdaf9)

---
Use the closed fireaxe Cabinet
![image](https://github.com/user-attachments/assets/e3b0e89b-9910-4c64-8786-bd6e9f66d69f)

---
Add a scrubber to this room since people are gonna be fuckin in here and they'll suffocate otherwise
![image](https://github.com/user-attachments/assets/4f9bbf59-e567-4707-a73e-dc458098f9ce)

---
![image](https://github.com/user-attachments/assets/28e788ad-9122-4267-8526-4202b6bdc053)
The light on the hidden door here is a big awkward and one of the two should be moved.

---
![image](https://github.com/user-attachments/assets/9e68e47c-3882-44d7-80dd-3d25d1b64b9f)
I suggest replacing this advanced mop with a regular mop. The advanced mop renders the mop bucket useless.

ItsDzordz replied:
the govles are there to ensure freelancers can use the vending machines in the kitchen, thats why they are there, since they dotn get the necessery service accceses. and well mercenary gloves that are insulated are part of the base load out. freelancer are not only miners and salvgers, they are everthing that doesnt fit into a mold or a job

Eagle0600 replied:
I would recommend *not* requiring someone's intended roundstart equipment to be hacked in order to be used as intended. Create a modified vending machine prototype locked to freelancer access instead.

southbridge-fur commented on Resources/Prototypes/_Floof/Access/freelancer.yml:
![image](https://github.com/user-attachments/assets/d59edc38-428a-476f-9fb2-b2280993784c)
Aghosts need this access too

ItsDzordz replied:
get a freelancer card, if i add the freelancer access to all access it will couse captain to be able to open and use ALL of the freelancer items

southbridge-fur replied:
https://github.com/Fansana/floofstation1/blob/fe9857a19235ae9dc4ed4b1f14e9bd255a3bdf7c/Resources/Prototypes/Entities/Mobs/Player/admin_ghost.yml#L39 just add the line for it here

southbridge-fur commented on Resources/Maps/Floof/Freelancer/Freelancer_shop:
I feel like you could color the pipes here too, especially since some of them are more exposed.

---

Name these buttons
![image](https://github.com/user-attachments/assets/6e779759-c1dd-4acc-83dd-6a8f2224d05b)

---
What is this janitorial light connected to? At first I thought it was like an "occupied" sign, but neither of the buttons nor the door toggle it.
![image](https://github.com/user-attachments/assets/bf255ce4-9765-4fc9-b4ae-7640cdc01d17)


---
Don't worry about these markers on the airlocks. I get it's to prevent someone from gasping for a moment as they cross, but it's not a big deal.
![image](https://github.com/user-attachments/assets/3d23bb0a-0f3f-40bb-8b0e-0d60f40d403e)

---
I feel like these are supposed to all be windoors so they're accessible
![image](https://github.com/user-attachments/assets/e151df8c-c4aa-4e84-883b-ad49cc1a73e7)

---
Pipe named Huge Guest Door
![image](https://github.com/user-attachments/assets/f4a02ece-be3b-43e0-8da1-b874270e3139)

---
Looks like this air alarm needs the vent and scrubbers added 
![image](https://github.com/user-attachments/assets/7ddff3e9-ccaa-4902-9a6a-2de08556a51a)

---
Why include non-functional cameras and two consoles which don't do anything?
![image](https://github.com/user-attachments/assets/b823ef4c-c554-419a-a8ae-b2fdab8aaae4)

ItsDzordz replied:
on the cmaeras they can be linked when the ship spawns, unforctuntley i have no clue how to do that while mapping

southbridge-fur replied:
If they're intended to be used, you'll need to use the regular camera entities and include a camera router on the ship (it's a server). There currently aren't cameras or a router for Freelancers but you could either add those or just use common

ItsDzordz replied:
oh i just relised i placed assemblies... and there is a rounter in engie room from what i rember. will check it out non the less

southbridge-fur commented on Resources/Maps/Floof/Freelancer/Freelancer_war_pod.yml:
I'd recommend making a fill for these rather than including all the items in them like this. It's better for portability. Just make a new `suit_storage.yml` in `Resources/Prototypes/_Floof/Catalog/Fills/Lockers` and copy one of these as your base https://github.com/Fansana/floofstation1/blob/aa8ec604bf779c1cb83a371d55adb67f8afc915d/Resources/Prototypes/Catalog/Fills/Lockers/suit_storage.yml

![image](https://github.com/user-attachments/assets/ea2870ee-71af-40c7-b6e6-73377c08e161)

---

Also like the other pod please don't stack the APC and substation.

ItsDzordz replied:
yeah for the lockers i didnt want to make filled lockers due to possibilty of item removal due to balancing of the items. can they stay like that for now, and i WILL be doing a part 2 to this pr, after i get some initial feedback form items and everything when its addded to the game

Eagle0600 replied:
Those balance changes can be made on the locker prototype you create.

southbridge-fur commented on Resources/Maps/Floof/Freelancer/Simulacrum_Module_base:
Was this supposed to be included?

ItsDzordz replied:
yes

southbridge-fur replied:
what it do?

ItsDzordz replied:
its a template arena ship for quick modifications by the admins

ItsDzordz replied:
the idea is that there is the main place for people to watch from and you can attach modules that you prepared on the side, or i will make some pre made ones in the future. this allows for a quick spawn of a figting arena without a need for building for a hour

southbridge-fur replied:
I'd suggest moving these to their own PR then.

ItsDzordz replied:
well if i move them i wont get fedback on the ship function and usfullness, it is ment to be admin spawn thing to save time in any case

southbridge-fur commented on Resources/Maps/Floof/Freelancer/Simulacrum_base:
Was this supposed to be included?

ItsDzordz replied:
yes

Commit:
Update jobs.yml

Commit:
Update access.yml

Commit:
Update access.yml

Commit:
Update airlocks.yml

Commit:
Update job.yml

ItsDzordz committed:
Ship fix

fixed ships as requested under supervision

southbridge-fur commented:
We covered all the requested changes in VC.

ItsDzordz committed:
Merge branch 'Protocol-3' of https://github.com/ItsDzordz/DzordzStuff into Protocol-3

southbridge-fur commented:
Alright well looks like there is an issue with `Simulacrum_Module_base`. I'm gonna ask you to remove these from the PR and shift them over to their own PR just so the freelancer stuff here can get merged. We can test the Simulacrum modules from the other PR.

ItsDzordz commented:
Removed simulacrum as requested

ItsDzordz committed:
Simulacrum removal

Mnemotechnician commented:
> The suits have weird 39.999999~% stamina damage reduction, which is similar to the Radiation suit's issue.

This is caused by the protection coefficient being irrepresentible with powers of 2 (0.6 cannot be represented as a sum of 2^n. You can get 0.5 + 0.0625 + ... and it adds up to slightly more than 0.6, which, when subtracted from 1, gives a little less than 0.4 - hence 0.39999...)

Not much we can really do without writing our own floating point formatting function - C# will automatically trim trailing zeros, but it cannot trim trailing nines.

ItsDzordz committed:
map invalid id fix

ItsDzordz commented:
i looked thru the test and it seems like one of the delta v salvge wrecks has something wrong, and small problems with other maps but not mine. but id advise somone who knows what they are doing to look thru it
```
 SERVER: 167.624s [INFO] loader: Loading Map: /Maps/Salvage/DeltaV/DV-cargo-01.yml
 SERVER: 167.627s [WARN] loader: Map contains an obsolete/removed prototype: SpawnVehicleATV. This may cause unexpected errors.
 SERVER: 167.627s [WARN] loader: Map contains an obsolete/removed prototype: VehicleKeyATV. This may cause unexpected errors.
```

Mnemotechnician commented:
> i looked thru the test and it seems like one of the delta v salvge wrecks has something wrong, and small problems with other maps but not mine. but id advise somone who knows what they are doing to look thru it

Nothing is wrong with delta-v maps, the error message states that ones of your maps (which one is unknown, RT doesn't tell, although ctrl-F search can tell you) contains a reference to a non-existent (perhaps deleted) entity. If you have linked linked one entity to another and then deleted the first entity, you must unlink the second before saving the map, otherwise it will lead to errors like this one.

ItsDzordz commented:
> > i looked thru the test and it seems like one of the delta v salvge wrecks has something wrong, and small problems with other maps but not mine. but id advise somone who knows what they are doing to look thru it
> 
> Nothing is wrong with delta-v maps, the error message states that ones of your maps (which one is unknown, RT doesn't tell, although ctrl-F search can tell you) contains a reference to a non-existent (perhaps deleted) entity. If you have linked linked one entity to another and then deleted the first entity, you must unlink the second before saving the map, otherwise it will lead to errors like this one.

damn i was extra carefull not to do anything like that, and it didnt pop up before last fix... seems i have to look into the last commit.

ItsDzordz commented:
![image](https://github.com/user-attachments/assets/d0995e7e-b567-411e-8070-1ff7e53acd1a) 
![image](https://github.com/user-attachments/assets/416f5a4e-5508-418d-8948-ed6b45980196)

ok i think it might be the shop, not shure if this line is the problem

southbridge-fur commented:
> ![image](https://private-user-images.githubusercontent.com/183029388/440131962-d0995e7e-b567-411e-8070-1ff7e53acd1a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDYyODk0NzgsIm5iZiI6MTc0NjI4OTE3OCwicGF0aCI6Ii8xODMwMjkzODgvNDQwMTMxOTYyLWQwOTk1ZTdlLWI1NjctNDExZS04MDcwLTFmZjdlNTNhY2QxYS5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwNTAzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDUwM1QxNjE5MzhaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1jMzZiMmI3ZDA3ZDA3MGRiNDI2MzU3ODFjOWRjYzJiMWU5ODY5M2NkNDQ4YzM5OTU0NDA2OGZlOTM3YWJkOGMzJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.GJJREbD6Ff15OBcfTL4Vykt-JNchOwb4v2utC7Ns79w) ![image](https://private-user-images.githubusercontent.com/183029388/440131997-416f5a4e-5508-418d-8948-ed6b45980196.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDYyODk0NzgsIm5iZiI6MTc0NjI4OTE3OCwicGF0aCI6Ii8xODMwMjkzODgvNDQwMTMxOTk3LTQxNmY1YTRlLTU1MDgtNDE4ZC04OTQ4LWVkNmI0NTk4MDE5Ni5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjUwNTAzJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI1MDUwM1QxNjE5MzhaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT02NTAzOWFiYzNlM2ExY2I5OTViYWFhNjE0OGNhZjBlNzA3ZjU2YzFiZGM4YzY2YzAxMzdlNWY5NmJmYTFlNDY3JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.12NjSuaNu41Jtkg9r2ar7tfmOV14tED6MTfhyvyT3Qg)
> 
> ok i think it might be the shop, not shure if this line is the problem

yeah that's it, just remove what you have highlighted there

ItsDzordz committed:
Shop ship bug fix

- removed cameras due it beeing prone to crashing and bugs
- removed couses of test fail
